### PR TITLE
Extend TestingApp with UI for visitor info update in acceptance tests

### PR DESCRIPTION
**Pull request by igorkravchenko**
Add UI that allows testing visitor info loading and updating visitor info in acceptance tests. At the moment error handling is not taken into account.

MOB-2339

### CODE REVIEW

File: TestingApp/Extensions/UIView+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        if let superview {
+            if predicate(superview) {
+                return superview
+            } else {
+                return superview.superview(by: predicate)
+            }
+        }
+
+        return nil
+    }
```
Comment (yurii-glia) [inline TestingApp/Extensions/UIView+Extensions.swift:31]: let's add some doc about this function and for what purposes it has been created.

File: TestingApp/VisitorInfo/VisitorInfoModel.swift [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+
+        return fields
+    }
+}
+
+extension VisitorInfoModel {
+    struct Environment {
+        let fetchVisitorInfo: (@escaping (Result<GliaCore.VisitorInfo, Error>) -> Void) -> Void
+        let updateVisitorInfo: (VisitorInfoUpdate, @escaping (Result<Bool, Error>) -> Void) -> Void
+        let uuid: () -> UUID
```
Comment (yurii-glia) [inline TestingApp/VisitorInfo/VisitorInfoModel.swift:435]: would we like to use Combine instead of functions here?

Comment (igorkravchenko) [reply TestingApp/VisitorInfo/VisitorInfoModel.swift:435]: Not sure if wrapping it into Combine publisher will be beneficial.
As side note: I tried to use Combine for reporting updated 'Props' using `combineLatest` but that did not work for me as I wanted to, so I left Combine out.

File: TestingApp/Extensions/UIView+Extensions.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            return true
+        }
+        for subview in view.subviews {
+            if hasResponder(subview) {
+                return true
+            }
+        }
+
+        return false
+    }
```
[the remark this answers is not in the document]
Comment (igorkravchenko) [reply TestingApp/Extensions/UIView+Extensions.swift]: Addressed 6eb28290fe9844fc6deacbeae58f42bed18df73f.

File: TestingApp/Extensions/UIView+Extensions.swift (hunk 1 above)
Comment (igorkravchenko) [reply TestingApp/Extensions/UIView+Extensions.swift:42]: Addressed 6eb28290fe9844fc6deacbeae58f42bed18df73f.

**Review by EgorovEI — APPROVED** [review]

**Review by rasmustautsglia — APPROVED** [review]

File: TestingApp/VisitorInfo/VisitorInfoViewController.HeaderView.swift [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+            titleLabel.topAnchor.constraint(equalTo: contentView.topAnchor, constant: margins),
+            titleLabel.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -margins),
+            stackView.centerYAnchor.constraint(equalTo: contentView.centerYAnchor),
+            stackView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -margins),
+        ])
+    }
+
+    func renderProps() {
+        titleLabel.text = props.title
+        renderedSegments = .init(segments: props.segments, selectedSegment: props.selectedSegment) //props.segments
```
Comment (gersonnoboa) [inline TestingApp/VisitorInfo/VisitorInfoViewController.HeaderView.swift:66]: Is there a reason for this comment?

**Review by gersonnoboa — APPROVED** [review]

File: TestingApp/VisitorInfo/VisitorInfoViewController.HeaderView.swift (hunk 4 above)
Comment (igorkravchenko) [reply TestingApp/VisitorInfo/VisitorInfoViewController.HeaderView.swift]: No, removed b9b42dcdc7b51038e214ce28b0e22876220c5bf0.

File: TestingApp/VisitorInfo/VisitorInfoViewController.Cells.swift [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        setup()
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) { fatalError("init(coder:) has not been implemented") }
+
+    private lazy var keyTextField = UITextField()
```
[the remark this answers is not in the document]
Comment (igorkravchenko) [reply TestingApp/VisitorInfo/VisitorInfoViewController.Cells.swift]: Totally agree with every statement. It was misplaced. 409e7857ff29b714035a43391ab434fe2f4c5d50

File: TestingApp/Extensions/UIView+Extensions.swift [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import UIKit
+
+extension UIView {
+    /// Checks if current view is firstResponder, if not check the same
+    /// for every subview. This is handy when we need to know if keyboard
+    /// is presented at the moment by the view or its subviews.
+    /// - Returns: `true` if view or subviews has `isFirstResponder` set to `true`.
+    func hasResponder() -> Bool {
```
[the remark this answers is not in the document]
Comment (igorkravchenko) [reply TestingApp/Extensions/UIView+Extensions.swift:8]: I think `isKeyboardPresented` fits well. Renamed 309be78918ca566cee03abd0f24882ea45970037.

**Comment by igorkravchenko** [comment]
!squash

File: TestingApp/VisitorInfo/VisitorInfoModel.swift [hunk 7]
```diff
@@ -0,0 +1,468 @@
...
+        didSet {
+            self.visitorInfoUpdate.customAttributesUpdateMethod = customAttributesUpdateMethod
+        }
+    }
+
+    var delegate: Delegate?
+
+    // Counts how many custom attributes have been created.
+    // Used to generate unique deterministic keys.
+    //var nextAttributeKeyCounter = 0
```
[the remark this answers is not in the document]
Comment (igorkravchenko) [reply TestingApp/VisitorInfo/VisitorInfoModel.swift]: No, removed it 9fb18d2278e9b4185f7ea5135bb2a199a13e9c97.

**Comment by igorkravchenko** [comment]
!squash

**Comment by igorkravchenko** [comment]
!squash

**Pull request merged**